### DISCUSSION TITLE
Problemi di connessione ad Home Assistant

### DISCUSSION THREAD

**Discussion by lnrd00** [indomus/forum#2877] [HUB: Home Assistant]
Salve, vado dritto al sodo: ho notato che quando provo ad accedere ad homeassistant da uno smartphone Android che non è mio (di mio padre), usando l'app Companion, risulta impossibile collegarsi e sui logs appare questa scritta:

```
WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from adsl-ull-246-87.49-151.wind.it (151.49.87.246). Requested URL: '/api/websocket'. (Mozilla/5.0 (Linux; Android 14; SM-G991B Build/UP1A.231005.007; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/120.0.6099.230 Mobile Safari/537.36 Home Assistant/2023.12.4-11898 (Android 14; SM-G991B))
```
Io, tra gli indirizzi IP specificati sul `configuration.yaml`, non ho effettivamente fatto rientrare IP come quello sopra indicato, cioè `151.49.87.246`, ma sia questo indirizzo IP che l'errore in generale sembra essere molto strano.
Nel mio `configuration.yaml` ho la seguente configurazione:

```
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 172.18.0.0/16
```
che finora ha sempre funzionato con tutti i device che uso per accedere, senza problemi.
Inoltre, se sempre dall'Android sopracitato provo ad accedere da browser attraverso l'indirizzo ip `http://INDIRIZZO_IP_DEL_RASPBERRY:8123` tutto funziona correttamente.

Qualcuno saprebbe indicarmi cosa devo fare per evitare che questo genere di situazione si ripresenti? Grazie in anticipo!

**Comment by snakuzzo** [comment]
> Nel mio configuration.yaml ho la seguente configurazione:

Non c'entra nulla la configurazione di `trusted_proxies` nel `configuration.yaml`.

> se sempre dall'Android sopracitato provo ad accedere da browser attraverso l'indirizzo ip http://INDIRIZZO_IP_DEL_RASPBERRY:8123 tutto funziona correttamente.

Eh...per forza...se vai diretto su HA non stai passando dal reverse proxy

> Qualcuno saprebbe indicarmi cosa devo fare per evitare che questo genere di situazione si ripresenti? Grazie in anticipo!

Verifica la configurazione del reverse proxy. Cosa stai utilizzando ?

**Reply by lnrd00** [reply]
Per il reverse proxy, ho seguito la guida inDomus e dunque utilizzo Caddy. Il `Caddyfile` è il seguente:

```
#  When your config is ok comment block below to disable use of AMCE staging server
#{
 #   acme_ca https://acme-staging-v02.api.letsencrypt.org/directory
#}
(https_header) {
  header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}
https://XXXXXX.duckdns.org {
  respond "Saluti dal Crew inDomus"
  file_server
}
https://XXXXXX.duckdns.org:9000 {
  import https_header
  reverse_proxy http://192.168.1.251:8123
}
```

**Reply by snakuzzo** [reply]
Prova a modificare il file `Caddyfile` aggiungendo questo alla configurazione del proxy per HA...

```
  @ws {
    header Connection *Upgrade*
    header Upgrade websocket
  }
```

Così...

```
#  When your config is ok comment block below to disable use of AMCE staging server
#{
#   acme_ca https://acme-staging-v02.api.letsencrypt.org/directory
#}

(https_header) {
  header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}

https://XXXXXX.duckdns.org {
  respond "Saluti dal Crew inDomus"
  file_server
}

https://XXXXXX.duckdns.org:9000 {
  import https_header
  reverse_proxy http://192.168.1.251:8123
  @ws {
    header Connection *Upgrade*
    header Upgrade websocket
  }
}
```

...restarta Caddy e vedi se funziona.

Fammi sapere

**Reply by lnrd00** [reply]
Funziona, grazie mille!! Potrei chiederti con esattezza cosa avrei appena inserito?

**Reply by snakuzzo** [reply]
Leggi qui -> https://companion.home-assistant.io/docs/troubleshooting/networking/#addendum-reverse-proxy-via-nginx

Nello specifico questa parte...

> _Note: If you don't use the NGINX Home Assistant add-on but instead roll your own, please ensure that websockets support is enabled_

Quello che ti ho fatto aggiungere consente la comunicazione con HA tramite [websocket](https://en.wikipedia.org/wiki/WebSocket). Nello specifico...

> _To achieve compatibility, the WebSocket handshake uses the [HTTP Upgrade header](https://en.wikipedia.org/wiki/HTTP/1.1_Upgrade_header) to change from the HTTP protocol to the WebSocket protocol._
